### ISSUE TITLE
Wrong SysConfig encoding breaks OTRS import

### ISSUE THREAD

**Issue by mastox** [zammad/zammad#1344]
### Infos:

* Used Zammad version:  1.5.0-1502623480.2d58abba.xenial
* Used Zammad installation source: package
* Operating system: Ubuntu 17.04
* Browser + version: Firefox-Dev 56.0b1
* OTRS-Repo: 5.0.28
* ZammadMigrator: 1.2.1
* OTRS 5.0.18

### Expected behavior:
Hey guys, i want to import data from otrs to zammad. I have tried to open all Request in Browser, and everything works fine. But Sysconfig gives me an 500 error.
Tried it also with a Docker Container but the same results.
Apache rewrites the http request to https. (on OTRS Server)
When i tried with https, i get only 500 errors.

Webimporter says: Background process did not start or has not finished! Please contact your support.

On every Import::OTRS.start it imports 2 Customers, but no Tickets.
Checked this on every import with:  User.find(1)...

Any Idea?

### Actual behavior:
```
> irb(main):001:0> Import::OTRS.start
thread#-: Start import...
thread#-: POST: http://HOST/otrs/public.pl?Action=ZammadMigrator
thread#-: PARAMS: {:Action=>"ZammadMigrator", :Key=>"API-KEY"}
thread#-: loading SysConfig...
thread#-: POST: http://HOST/otrs/public.pl?Action=ZammadMigrator
thread#-: PARAMS: {:Subaction=>"Export", :Object=>"SysConfig", :Limit=>"", :Offset=>"", :Diff=>0, :Action=>"ZammadMigrator", :Key=>"API-KEY"}
thread#-: ... no more work.
thread#-: loading DynamicField...
thread#-: POST: http://HOST/otrs/public.pl?Action=ZammadMigrator
thread#-: PARAMS: {:Subaction=>"Export", :Object=>"DynamicField", :Limit=>"", :Offset=>"", :Diff=>0, :Action=>"ZammadMigrator", :Key=>"API-KEY"}
thread#-: ... no more work.
thread#-: loading State...
thread#-: POST: http://HOST/otrs/public.pl?Action=ZammadMigrator
thread#-: PARAMS: {:Subaction=>"Export", :Object=>"State", :Limit=>"", :Offset=>"", :Diff=>0, :Action=>"ZammadMigrator", :Key=>"API-KEY"}
thread#-: ... no more work.
thread#-: loading Priority...
thread#-: POST: http://HOST/otrs/public.pl?Action=ZammadMigrator
thread#-: PARAMS: {:Subaction=>"Export", :Object=>"Priority", :Limit=>"", :Offset=>"", :Diff=>0, :Action=>"ZammadMigrator", :Key=>"API-KEY"}
thread#-: ... no more work.
thread#-: loading Queue...
thread#-: POST: http://HOST/otrs/public.pl?Action=ZammadMigrator
thread#-: PARAMS: {:Subaction=>"Export", :Object=>"Queue", :Limit=>"", :Offset=>"", :Diff=>0, :Action=>"ZammadMigrator", :Key=>"API-KEY"}
thread#-: ... no more work.
thread#-: loading User...
thread#-: POST: http://HOST/otrs/public.pl?Action=ZammadMigrator
thread#-: PARAMS: {:Subaction=>"Export", :Object=>"User", :Limit=>"", :Offset=>"", :Diff=>0, :Action=>"ZammadMigrator", :Key=>"API-KEY"}
thread#-: ... no more work.
thread#-: loading Customer...
thread#-: POST: http://HOST/otrs/public.pl?Action=ZammadMigrator
thread#-: PARAMS: {:Subaction=>"Export", :Object=>"Customer", :Limit=>"", :Offset=>"", :Diff=>0, :Action=>"ZammadMigrator", :Key=>"API-KEY"}
thread#-: ... no more work.
thread#-: Importing CustomerUser in steps of 50
thread#-: loading 50 CustomerUser starting at 0...
thread#-: POST: http://HOST/otrs/public.pl?Action=ZammadMigrator
thread#-: PARAMS: {:Subaction=>"Export", :Object=>"CustomerUser", :Limit=>50, :Offset=>0, :Diff=>0, :Action=>"ZammadMigrator", :Key=>"API-KEY"}
thread#-: ... no more work.
thread#7: Importing Ticket in steps of 20
thread#7: loading 20 Ticket starting at 120...
thread#7: POST: http://HOST/otrs/public.pl?Action=ZammadMigrator
thread#7: PARAMS: {:Subaction=>"Export", :Object=>"Ticket", :Limit=>20, :Offset=>120, :Diff=>0, :Action=>"ZammadMigrator", :Key=>"API-KEY"}
thread#1: Importing Ticket in steps of 20
thread#1: loading 20 Ticket starting at 0...
thread#1: POST: http://HOST/otrs/public.pl?Action=ZammadMigrator
thread#1: PARAMS: {:Subaction=>"Export", :Object=>"Ticket", :Limit=>20, :Offset=>0, :Diff=>0, :Action=>"ZammadMigrator", :Key=>"API-KEY"}
thread#5: Importing Ticket in steps of 20
thread#5: loading 20 Ticket starting at 80...
thread#5: POST: http://HOST/otrs/public.pl?Action=ZammadMigrator
thread#5: PARAMS: {:Subaction=>"Export", :Object=>"Ticket", :Limit=>20, :Offset=>80, :Diff=>0, :Action=>"ZammadMigrator", :Key=>"API-KEY"}
thread#4: Importing Ticket in steps of 20
thread#4: loading 20 Ticket starting at 60...
thread#4: POST: http://HOST/otrs/public.pl?Action=ZammadMigrator
thread#4: PARAMS: {:Subaction=>"Export", :Object=>"Ticket", :Limit=>20, :Offset=>60, :Diff=>0, :Action=>"ZammadMigrator", :Key=>"API-KEY"}
thread#6: Importing Ticket in steps of 20
thread#6: loading 20 Ticket starting at 100...
thread#6: POST: http://HOST/otrs/public.pl?Action=ZammadMigrator
thread#3: Importing Ticket in steps of 20
thread#6: PARAMS: {:Subaction=>"Export", :Object=>"Ticket", :Limit=>20, :Offset=>100, :Diff=>0, :Action=>"ZammadMigrator", :Key=>"API-KEY"}
thread#3: loading 20 Ticket starting at 40...
thread#2: Importing Ticket in steps of 20
thread#3: POST: http://HOST/otrs/public.pl?Action=ZammadMigrator
thread#2: loading 20 Ticket starting at 20...
thread#3: PARAMS: {:Subaction=>"Export", :Object=>"Ticket", :Limit=>20, :Offset=>40, :Diff=>0, :Action=>"ZammadMigrator", :Key=>"API-KEY"}
thread#2: POST: http://HOST/otrs/public.pl?Action=ZammadMigrator
thread#8: Importing Ticket in steps of 20
thread#2: PARAMS: {:Subaction=>"Export", :Object=>"Ticket", :Limit=>20, :Offset=>20, :Diff=>0, :Action=>"ZammadMigrator", :Key=>"API-KEY"}
thread#8: loading 20 Ticket starting at 140...
thread#8: POST: http://HOST/otrs/public.pl?Action=ZammadMigrator
thread#8: PARAMS: {:Subaction=>"Export", :Object=>"Ticket", :Limit=>20, :Offset=>140, :Diff=>0, :Action=>"ZammadMigrator", :Key=>"API-KEY"}
thread#7: ... no more work.
thread#4: ... no more work.
thread#2: ... no more work.
thread#1: ... no more work.
thread#5: ... no more work.
thread#8: ... no more work.
thread#6: ... no more work.
thread#3: ... no more work.
```

### Steps to reproduce the behavior:

* 
Setting.set('import_otrs_endpoint', 'http://xxx/otrs/public.pl?Action=ZammadMigrator')
Setting.set('import_otrs_endpoint_key', 'xxx')
Setting.set('import_mode', true)
Import::OTRS.start

**Comment by mastox** [comment]
Hi @thorsteneckel ,
u mean this logfile /opt/otrs/var/log ?
It only say's that i have an Invalid API-Key.
But when i open this in Browser with that api key everything works fine.
Also tried to set a new API Key. 
[otrs.txt](https://github.com/zammad/zammad/files/1222327/otrs.txt)

**Comment by thorsteneckel** [comment]
It depends on how you configured your OTRS. Please check the SysConfig `LogModule`. If you your configuration is `SysLog` it will log to your central OS log file. This is usually `/var/log/messages`. If you have configured a custom file, check this instead.

The wrong API Key should not fail with a 500 status code.

Please provide the full error information about the 500 error.
Just to make sure: Please provide a full list of installed packages.

**Comment by mastox** [comment]
We are logging into this file, 
` /opt/otrs/var/log`
put theres only this message with the api key.
but i found this in the apache log

```
ERROR: OTRS-CGI-10 Perl: 5.22.1 OS: linux Time: Mon Aug 14 16:00:18 2017'
 
 Message: Zammad Migrator: Invalid API key.
 
 RemoteAddress: 10.0.0.6
 RequestURI: /otrs/public.pl?Action=ZammadMigrator
 
 Traceback (19034): 
   Module: Kernel::Modules::ZammadMigrator::_Error Line: 158
   Module: Kernel::Modules::ZammadMigrator::Run Line: 50
   Module: Kernel::System::Web::InterfacePublic::Run Line: 220
   Module: ModPerl::ROOT::ModPerl::Registry::opt_otrs_bin_cgi_2dbin_public_2epl::handler Line: 40
   Module: (eval) (v1.99) Line: 207
   Module: ModPerl::RegistryCooker::run (v1.99) Line: 207
   Module: ModPerl::RegistryCooker::default_handler (v1.99) Line: 173
   Module: ModPerl::Registry::handler (v1.99) Line: 32

**Comment by hanneshal** [comment]
@mastox Can you check the Apache Error Log for your host. The 500er should be there too. This small snippet is sadly not enough

**Comment by mastox** [comment]
@hanneshal We deleted all log files, before we started to migrate, there are only this lines in the file, 10 times.
So i tried to start migrate again and found this interesting line in error.log
Maybe this could be the problem ?

Everytime i push the button migrate from orts there a new line ....

```

[Wed Aug 16 14:05:45.767330 2017] [:error] [pid 9822] malformed or illegal unicode character in string [\xdcber folgen], cannot convert to JSON at /opt/otrs/Kernel/System/JSON.pm line 96.\n
[Wed Aug 16 14:06:21.850823 2017] [:error] [pid 8803] malformed or illegal unicode character in string [\xdcber folgen], cannot convert to JSON at /opt/otrs/Kernel/System/JSON.pm line 96.\n
[Wed Aug 16 14:06:21.851163 2017] [:error] [pid 8699] malformed or illegal unicode character in string [\xdcber folgen], cannot convert to JSON at /opt/otrs/Kernel/System/JSON.pm line 96.\n
[Wed Aug 16 14:06:30.800693 2017] [:error] [pid 9819] malformed or illegal unicode character in string [\xdcber folgen], cannot convert to JSON at /opt/otrs/Kernel/System/JSON.pm line 96.\n
[Wed Aug 16 14:07:14.242951 2017] [:error] [pid 9539] malformed or illegal unicode character in string [\xdcber folgen], cannot convert to JSON at /opt/otrs/Kernel/System/JSON.pm line 96.\n
[Wed Aug 16 14:07:29.393102 2017] [:error] [pid 9539] malformed or illegal unicode character in string [\xdcber folgen], cannot convert to JSON at /opt/otrs/Kernel/System/JSON.pm line 96.\n
[Wed Aug 16 14:07:30.142749 2017] [:error] [pid 9539] malformed or illegal unicode character in string [\xdcber folgen], cannot convert to JSON at /opt/otrs/Kernel/System/JSON.pm line 96.\n
[Wed Aug 16 14:07:30.695429 2017] [:error] [pid 9539] malformed or illegal unicode character in string [\xdcber folgen], cannot convert to JSON at /opt/otrs/Kernel/System/JSON.pm line 96.\n
[Wed Aug 16 14:07:31.193996 2017] [:error] [pid 9819] malformed or illegal unicode character in string [\xdcber folgen], cannot convert to JSON at /opt/otrs/Kernel/System/JSON.pm line 96.\n
[Wed Aug 16 14:07:44.279015 2017] [:error] [pid 8650] malformed or illegal unicode character in string [\xdcber folgen], cannot convert to JSON at /opt/otrs/Kernel/System/JSON.pm line 96.\n

```

**Comment by mastox** [comment]
We installed a new OTRS with the same version like ours 5.0.18 and tried to migrate data to zammad.
Everything works fine, but if we add a new ticket we get the following error's

[apache_error.txt](https://github.com/zammad/zammad/files/1234220/apache_error.txt)

[api_call.txt](https://github.com/zammad/zammad/files/1234213/api_call.txt)

**Comment by mastox** [comment]
Now it works!
In the File ` /opt/otrs/Kernel/Config/Files/ZZZAuto.pm ` there was some 'ä ü ö' the encoding of these charakters was in Windows-1252 saved, but on top of the file is a 
` use utf8;`
I changed this charackters ans saved the file as UTF-8.

**Comment by thorsteneckel** [comment]
Nice! Thanks for the feedback. Closing now.
